docs/source/proposals/external-memory-management.rst

```diff
@@ -0,0 +1,918 @@
+           # the RMM C++ layer for initialising IPC handles, and not use any Numba
+           # internals.
+           ipchandle = (ctypes.c_byte * 64)()  # IPC handle is 64 bytes
+           cuda.cudadrv.memory.driver_funcs.cuIpcGetMemHandle(
+               ctypes.byref(ipchandle),
+               memory.owner.handle,
+           )
+           source_info = cuda.current_context().device.get_device_identity()
+           ptr = memory.device_ctypes_pointer.value
+           offset = librmm.rmm_getallocationoffset(ptr, 0)
+           return IpcHandle(memory, ipchandle, memory.size, source_info,
+                            offset=offset)
+
+       def get_memory_info(self):
+           # Returns a tuple of (free, total) using RMM functionality.
+           return get_memory_info()
+
+       def initialize(self):
+           # Nothing required to initialize RMM here, but this method is added
+           # to illustrate that the super() method should also be called.
+           super().initialize()
+
+       @contextmanager
+       def defer_cleanup(self):
+           # Does nothing to defer cleanup - a full implementation may choose to
+           # implement a different policy.
+           with super().defer_cleanup():
+               yield
+
+       @property
+       def interface_version(self):
+           # As required by the specification
+           return 1
+
+   # The existing _make_finalizer function is used by RMMNumbaManager:
+   def _make_finalizer(handle, stream):
+       """
+       Factory to make the finalizer function.
+       We need to bind *handle* and *stream* into the actual finalizer, which
+       takes no args.
+       """
+
+       def finalizer():
+           """
+           Invoked when the MemoryPointer is freed
+           """
+           librmm.rmm_free(handle, stream)
+
+       return finalizer
+
+   # Utility function register `RMMNumbaManager` as an EMM:
+   def use_rmm_for_numba():
+       cuda.set_memory_manager(RMMNumbaManager)
+
+   # To support `NUMBA_CUDA_MEMORY_MANAGER=rmm`:
+   _numba_memory_manager = RMMNumbaManager
+
+Example usage
+^^^^^^^^^^^^^
+
+A simple example that configures Numba to use RMM for memory management and
+creates a device array is as follows:
+
+.. code-block:: python
+
+   # example.py
+   import rmm 
+   import numpy as np
+
+   from numba import cuda
+
+   rmm.use_rmm_for_numba()
+
+   a = np.zeros(10)
+   d_a = cuda.to_device(a)
+   del(d_a)
+   print(rmm.csv_log())
+
+Running this should result in output similar to the following:
+
+.. code-block::
+
+   Event Type,Device ID,Address,Stream,Size (bytes),Free Memory,Total Memory,Current Allocs,Start,End,Elapsed,Location
+   Alloc,0,0x7fae06600000,0,80,0,0,1,1.10549,1.1074,0.00191666,/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py:683
+   Free,0,0x7fae06600000,0,0,0,0,0,1.10798,1.10921,0.00122238,/home/nfs/gmarkall/numbadev/numba/numba/utils.py:678
```

Comment:
Is there a reason to do this? I was trying to give a representative example of output from running with the external memory manager - for the documentation, I can imagine paths might be a little less desirable, but for the NBEP document I'm not sure I understand why it would matter.

Reply:
Largely just consistency with elsewhere. `egrep -r 'path>' docs/source/`.

Reply:
Paths now anonymised, replaced with `<path>`.